physionet-django/tests.py

```diff
@@ -0,0 +1,67 @@
+from schema import Query
+from django.test.testcases import TestCase
+import graphene
+import pdb
+
+
+class AnExampleTest(TestCase):
+
+    def test_all_published_projects(self):
+        schema = graphene.Schema(query=Query, auto_camelcase=False)
+        query = """
+            query {
+                all_published_projects {
+                    title
+                    version
+                    slug
+                    abstract
+                    main_storage_size
+                    compressed_storage_size
+                }
+            }
+        """
+        result = schema.execute(query)
+        self.assertIsNone(result.errors)
+
+    def test_all_published_authors(self):
+        schema = graphene.Schema(query=Query, auto_camelcase=False)
+        query = """
+            query {
+                all_published_authors {
+                    first_names
+                    last_name
+                    corresponding_email
+                    project {
+                        title
+                        version
+                        slug
+                        abstract
+                        main_storage_size
+                        compressed_storage_size
+                    }
+                }
+            }
+        """
+        result = schema.execute(query)
+        self.assertIsNone(result.errors)
+
+    def test_all_published_references(self):
+        schema = graphene.Schema(query=Query, auto_camelcase=False)
+        query = """
+            query {
+                all_published_references {
+                    id
+                    description
+                    project {
+                        title
+                        version
+                        slug
+                        abstract
+                        main_storage_size
+                        compressed_storage_size
+                    }
+                }
+            }
+        """
+        result = schema.execute(query)
```

Comment:
Can we check that the results are expected with a comparison to the known content?

Reply:
Yes actually, we should be able to extract the content with `json.dumps(result.content)` I believe and compare it with either the fixtures or directly from the project objects.

Reply:
the fixtures are loaded by `loaddemo` so comparisons can be made to the project object.

```py
project = PublishedProject.objects.get(title="thetitle")
assertequals(project.abstract,abstract)
```